sql/sql_parse.cc

```diff
@@ -1164,6 +1164,14 @@ static bool wsrep_tables_accessible_when_detached(const TABLE_LIST *tables)
   }
   return true;
 }
+
+static bool wsrep_command_no_result(char command)
+{
+  return (/* command == COM_STMT_PREPARE          || */
```

Comment:
It should be investigated if `COM_STMT_PREPARE` should be included in this condition or not.

Reply:
28ad6a7f28d1bdfb0d10530f8ab667213005807a
e996304189999eae8aeaa6995a80bf486833ed87
c18e0dab8a24db9d8f84fef328c27a28939a6ef5

Reply:
As you mentioned, it might be confusing the to the client to find a deadlock error during stmt prepare. Moreover, it has been like this for a long time and I see no real benefit to change this now.
I included COM_STMT_PREPARE in wsrep_command_no_result().